xlive/XLive/Networking/xlive_network.cpp

```diff
@@ -262,25 +261,17 @@ int WINAPI XSocketRecvFrom(SOCKET s, char *buf, int len, int flags, sockaddr *fr
 		u_long iplong = (((struct sockaddr_in*)from)->sin_addr.s_addr);
 		std::pair <ULONG, SHORT> hostpair = std::make_pair(iplong, port);
 
+		SecurePacket* secure_pck = reinterpret_cast<SecurePacket*>(buf);
 		if (iplong != H2Config_master_ip)
 		{
-			if (*(ULONG*)buf == annoyance_factor && ret == 12 + sizeof(XNADDR))
+			if (ret == sizeof(SecurePacket)
```

Comment:
would be nicer to call it something other than `ret`, maybe `packet_size`. Could also merge the two if statements.

Reply:
the if statements can't be merged

Reply:
I missed the rest of the code that handles non-secure packets